### PR TITLE
Add CDK for deployment

### PR DESCRIPTION
**Pull request by harrisonhjones**
# Overview

This code changes adds a cdk project for semi-automated deployment.

# Testing

1. Followed the README in `cdk/README.md`.
1. Look at that deployed app: https://dtrpppgacyhz2.cloudfront.net

### CODE REVIEW

**Comment by harrisonhjones** [comment]
Based on https://medium.com/swlh/aws-cdk-and-typescript-deploy-a-static-react-app-to-s3-df74193e9e3d

File: cdk/lib/cdk-stack.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    // S3:
+    const bucket = new s3.Bucket(this, "Bucket", {
+      // blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+
+    // Deployment:
+    const src = new s3Deploy.BucketDeployment(this, "Deployment", {
+      sources: [s3Deploy.Source.asset("../build")],
```
Comment (TheYeung1) [inline cdk/lib/cdk-stack.ts:18]: I'm guessing this will be the actual code for our react application after we run `npm build`?

**Review by TheYeung1 — APPROVED** [review]

File: cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Setup
+
+1. If you haven't bootstrapped a CDK project in your AWS account:
+   `cdk bootstrap`.
+
+# Development & Deployment
+
+## Development
+
+1. Make changes to `.ts` files; `.js` files are generated.
```
[the remark this answers is not in the document]
Comment (harrisonhjones) [reply cdk/README.md:14]: I was trying to say that you should only modify `.ts` files. The `.js` files are generated during build from the `.ts` files.

File: cdk/lib/cdk-stack.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import * as s3 from "@aws-cdk/aws-s3";
+import * as s3Deploy from "@aws-cdk/aws-s3-deployment";
+
+export class CdkStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // S3:
+    const bucket = new s3.Bucket(this, "Bucket", {
+      // blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
```
[the remark this answers is not in the document]
Comment (harrisonhjones) [reply cdk/lib/cdk-stack.ts:12]: Yea. Will do

File: cdk/lib/cdk-stack.ts (hunk 1 above)
Comment (harrisonhjones) [reply cdk/lib/cdk-stack.ts:18]: Correct

File: cdk/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (harrisonhjones) [reply cdk/README.md:14]: I've expanded the comment to make this more clear.

**Pull request merged**